src/components/swap/UnsupportedCurrencyFooter.test.tsx

```diff
@@ -0,0 +1,74 @@
+import userEvent from '@testing-library/user-event'
+import { Token } from '@uniswap/sdk-core'
+import { useUnsupportedTokens } from 'hooks/Tokens'
+import { mocked } from 'test-utils/mocked'
+import { fireEvent, render, screen, waitForElementToBeRemoved, within } from 'test-utils/render'
+import { getExplorerLink } from 'utils/getExplorerLink'
+
+import UnsupportedCurrencyFooter from './UnsupportedCurrencyFooter'
+
+const unsupportedTokenAddress = '0x4e83b6287588a96321B2661c5E041845fF7814af'
+const unsupportedTokenSymbol = 'ALTDOM-MAR2021'
+const unsupportedToken = new Token(1, unsupportedTokenAddress, 18, 'ALTDOM-MAR2021')
+const unsupportedTokenExplorerLink = 'www.blahblah.com'
+
+jest.mock('../../hooks/Tokens')
+
+jest.mock('../../utils/getExplorerLink')
+
+describe('UnsupportedCurrencyFooter.tsx with unsupported tokens', () => {
+  beforeEach(() => {
+    mocked(useUnsupportedTokens).mockImplementation(() => ({ [unsupportedTokenAddress]: unsupportedToken }))
+    mocked(getExplorerLink).mockImplementation(() => unsupportedTokenExplorerLink)
```

Comment:
This can be simplified:

```suggestion
    mocked(useUnsupportedTokens).mockReturnsValue({ [unsupportedTokenAddress]: unsupportedToken })
    mocked(getExplorerLink).mockReturnsValue(unsupportedTokenExplorerLink)
```